lib/react_client.dart

```diff
@@ -634,7 +634,11 @@ abstract class _ReactDartInteropStatics2 {
         try {
           throwErrorFromJS(error);
         } catch (e) {
-          return jsBackingMapOrJsCopy(componentStatics.instanceForStaticMethods.getDerivedStateFromError(e));
+          dynamic result = componentStatics.instanceForStaticMethods.getDerivedStateFromError(e);
+          if (result != null) {
+            return jsBackingMapOrJsCopy(result);
+          }
```

Comment:
#nit this could be a one liner
`
if (result != null) return jsBackingMapOrJsCopy(result);
`